ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -203,20 +215,30 @@ def _create_model_path(model_path):
                 "permissions are set correctly.".format(model_path)
             )
 
+    @staticmethod
+    def _check_reset_params(reset_params, new_config):
+        for k in new_config:
+            if (k in reset_params) and (isinstance(config[k], (int, float))):
+                continue
+            elif not isinstance(new_config[k], (int, float)):
+                raise UnityEnvironmentException(
+                    "The parameter '{0}'' generated by the sampler doesn't exist in this environment.".format(
+                        k
+                    )
+                )
+
     def _reset_env(self, env: BaseUnityEnvironment):
         """Resets the environment.
 
         Returns:
             A Data structure corresponding to the initial reset state of the
             environment.
         """
-        if self.meta_curriculum is not None:
-            return env.reset(
-                train_mode=self.fast_simulation,
-                config=self.meta_curriculum.get_config(),
-            )
-        else:
-            return env.reset(train_mode=self.fast_simulation)
+        sampled_reset_param = self.sampler_manager.sample_all()
```

Comment:
Does this still work if sampler_manager is empty?

Reply:
Ensured that this works if sampler_manager is empty; part of the sampler_manager test to check if this is being done